mmedit/models/editors/guided_diffusion/adm.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from copy import deepcopy
+from typing import List, Optional
+
+import mmengine
+import torch
+from mmengine import MessageHub
+from mmengine.model import BaseModel, is_model_wrapper
+from mmengine.optim import OptimWrapperDict
+from mmengine.runner.checkpoint import _load_checkpoint_with_prefix
+from tqdm import tqdm
+
+from mmedit.registry import DIFFUSERS, MODELS, MODULES
+from mmedit.structures import EditDataSample, PixelData
+from mmedit.utils.typing import ForwardInputs, SampleList
+
+def classifier_grad(x, t, y=None, classifier_scale=1.0):
```

Comment:
Since the guidance function is separate from `ADM`, why not change this function to a registered class and support users to define their own guidance function by config?